packages/core/src/index.js

```diff
@@ -330,23 +330,19 @@ const createCss = (init) => {
 	}
 
 	const css = (...inits) => {
-		let type
 		let composers = []
 		let composer
 		let defaultVariants = create(null)
 
 		for (const init of inits) {
 			if ($$composers in Object(init)) {
-				type = init.type || type
 				for (const composer of init[$$composers]) {
 					composers.push(composer)
 					assign(defaultVariants, composer.defaultVariants)
 				}
-			} else if (init && typeof init === 'object' && !('type' in init)) {
+			} else if (init && typeof init === 'object') {
 				composers.push((composer = createComposer(init)))
 				assign(defaultVariants, composer.defaultVariants)
-			} else {
-				type = ('type' in Object(init) ? init.type : init) || type
```

Comment:
This is a difficult problem, and I’ve been thinking about it.

This wasn’t here _just_ for React. It was the catch-all for any type-like value, including functions (which can also be components) or anything that didn’t seem like a component.

I think, to really pull this from core, a function similar to this needs to exist in react.